server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -564,6 +572,13 @@ IndexShardStats indexShardStats(final IndicesService indicesService, final Index
         );
     }
 
+    private Map<Index, ProjectId> projectsByIndex() {
+        return indices.values()
+            .stream()
+            .map(AbstractIndexComponent::index)
+            .collect(Collectors.toMap(index -> index, index -> clusterService.state().metadata().projectFor(index).id()));
```

Comment:
I'm wondering if it's worth it to build this map manually instead of using Streams, for performance reasons. I believe the stats API is called pretty regularly by monitoring agents (but I could be mistaken), and if we're doing this for all indices in all projects, the difference between Streams and non-streams might be noticeable. What do you think?